tools/doc/allhtml.js

```diff
@@ -73,3 +73,16 @@ all = all.slice(0, apiStart.index + apiStart[0].length) +
 
 // Write results.
 fs.writeFileSync(source + '/all.html', all, 'utf8');
+
+// Validate all hrefs have a target.
+const ids = new Set();
+const id_re = / id="(\w+)"/g;
+let match;
+while (match = id_re.exec(all)) {
+  ids.add(match[1]);
+}
+
+const href_re = / href="#(\w+)"/g;
+while (match = href_re.exec(all)) {
+  if (!ids.has(match[1])) throw new Error(`link not found: ${match[1]}`);
```

Comment:
Maybe it would be a bit more efficient if we cache `match[1]` in a variable here to not retrieve it twice?